desktop-widgets/mainwindow.cpp

```diff
@@ -1835,9 +1822,15 @@ void MainWindow::showFilterIfEnabled()
 		ui.bottomSplitter->setSizes({ EXPANDED, COLLAPSED });
 	}
 }
-void MainWindow::registerApplicationState(ApplicationState state, QWidget *topLeft, QWidget *topRight, QWidget *bottomLeft, QWidget *bottomRight)
+void MainWindow::registerApplicationState(ApplicationState state, QWidget *topLeft, QWidget *topRight, QWidget *bottomLeft, QWidget *bottomRight,
+					  int topLeftFlags, int topRightFlags, int bottomLeftFlags, int bottomRightFlags)
 {
-	applicationState[(int)state] = WidgetForQuadrant(topLeft, topRight, bottomLeft, bottomRight);
+	applicationState[(int)state] = {
+		{ topLeft, topLeftFlags},
+		{ topRight, topRightFlags},
+		{ bottomLeft, bottomLeftFlags},
+		{ bottomRight, bottomRightFlags}
```

Comment:
And this is why I thought the first order was better... because we use them in these pairs...